testsuite/manualmode/src/test/java/org/wildfly/core/test/standalone/mgmt/ManagementMultipartRequestTestCase.java

```diff
@@ -0,0 +1,410 @@
+        authCache.put(new HttpHost(managementUri.getHost()), digestScheme);
+
+        final HttpClientContext context = HttpClientContext.create();
+        context.setCredentialsProvider(credsProvider);
+        context.setAuthCache(authCache);
+
+        // request
+        try {
+            HttpResponse response = client.execute(post, context);
+            Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+        } finally {
+            ((CloseableHttpClient) client).close();
+        }
+    }
+
+    private static ContentBody createDeploymentOperationPart() throws IOException {
+        ModelNode op = Util.createAddOperation(PathAddress.pathAddress(DEPLOYMENT, DEPLOYMENT_NAME));
+        op.get("enabled").set(true);
+        op.get("content").add().get(INPUT_STREAM_INDEX).set(0);
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        op.writeBase64(baos);
+
+        return new ByteArrayBody(baos.toByteArray(), ContentType.create("application/dmr-encoded"), "blob");
+    }
+
+    private static ContentBody createDeploymentContentPart() {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        WebArchive war = createDeployment();
+        war.as(ZipExporter.class).exportTo(baos);
+        LOG.infof("Deployment size is %d bytes", baos.size());
+        Assert.assertTrue("Generated war is too small, add some more classes!", baos.size() > 20000);
+
+        return new ByteArrayBody(baos.toByteArray(), ContentType.create("application/x-webarchive"), "deployment.war");
+    }
+
+    private static WebArchive createDeployment() {
+        // include whatever classes, but need to reach certain minimal size
+        return ShrinkWrap.create(WebArchive.class, DEPLOYMENT_NAME)
+                .addPackage(ShrinkWrap.class.getPackage());
+    }
+
+    private static void keyMaterialSetup() throws Exception {
+        FileUtils.deleteDirectory(WORK_DIR);
+        Assert.assertTrue(WORK_DIR.mkdirs());
+        CoreUtils.createKeyMaterial(WORK_DIR);
+    }
+
+    private static void createManagementUsers() throws NoSuchAlgorithmException, IOException {
+        String passwordHash = new UsernamePasswordHashUtil()
+                .generateHashedHexURP(USERNAME, MANAGEMENT_REALM, PASSWORD.toCharArray());
+        String line = String.format("\n%s=%s\n", USERNAME, passwordHash);
+        Files.write(MGMT_USERS_PROPERTY_PATH, line.getBytes(), StandardOpenOption.TRUNCATE_EXISTING);
+    }
+
+    private static void deleteManagementUsers() throws IOException {
+        Files.write(MGMT_USERS_PROPERTY_PATH, "\n".getBytes(), StandardOpenOption.TRUNCATE_EXISTING);
+    }
+
+    private static String getServerAddress() throws UnknownHostException {
+        return InetAddress.getByName(TestSuiteEnvironment.getServerAddress()).getHostAddress();
+    }
+
+    private static void executeReloadAndWaitForCompletion(ModelControllerClient client, ModelControllerClient afterReloadClient) {
```

Comment:
why did you not use the methods from org.jboss.as.test.shared.ServerReload to wait for the live server to reload?

Reply:
I did not use it because I needed the waiting logic after reload to use different interface and protocol (remoting over native). When looking again I found that there's a method for this too in ServerReload which I originally overlooked, so I'm switching to that. Thanks.